File: WCF Data Services Quickstart (OData Service and WPF Client)/[C#]-WCF Data Services Quickstart (OData Service and WPF Client)/C#/NorthwindService/Northwind.svc.cs
using System;
using System.Collections.Generic;
using System.Data.Services;
using System.Linq;
using System.ServiceModel.Web;
using System.Web;

namespace NorthwindService
{
    public class Northwind : DataService<NorthwindEntities>
    {
        // This method is called only once to initialize service-wide policies.
        public static void InitializeService(IDataServiceConfiguration config)
        {
            // Make certain entity sets writable.
            config.SetEntitySetAccessRule("Customers", EntitySetRights.All);
            config.SetEntitySetAccessRule("Employees", EntitySetRights.All);
            config.SetEntitySetAccessRule("Orders", EntitySetRights.All);
            config.SetEntitySetAccessRule("Order_Details", EntitySetRights.All);
            config.SetEntitySetAccessRule("Products", EntitySetRights.All);

            // Make the remaining entity sets read-only.
            config.SetEntitySetAccessRule("*", EntitySetRights.AllRead);
        }
    }
}

File: WCF Data Services Quickstart (OData Service and WPF Client)/[VB]-WCF Data Services Quickstart (OData Service and WPF Client)/VB/NorthwindService/Northwind.svc.vb
Imports System.Data.Services
Imports System.Linq
Imports System.ServiceModel.Web

Public Class Northwind
    Inherits DataService(Of NorthwindEntities)

    ' This method is called only once to initialize service-wide policies.
    Public Shared Sub InitializeService(ByVal config As IDataServiceConfiguration)
        ' Make certain entity sets writable.
        config.SetEntitySetAccessRule("Customers", EntitySetRights.All)
        config.SetEntitySetAccessRule("Employees", EntitySetRights.All)
        config.SetEntitySetAccessRule("Orders", EntitySetRights.All)
        config.SetEntitySetAccessRule("Order_Details", EntitySetRights.All)
        config.SetEntitySetAccessRule("Products", EntitySetRights.All)

        ' Make the remaining entity sets read-only.
        config.SetEntitySetAccessRule("*", EntitySetRights.AllRead)
    End Sub

End Class
